ui/tests/integration/components/calendar-widget-test.js

```diff
@@ -0,0 +1,123 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click } from '@ember/test-helpers';
+import sinon from 'sinon';
+import hbs from 'htmlbars-inline-precompile';
+import calendarDropdown from 'vault/tests/pages/components/calendar-widget';
+
+module('Integration | Component | calendar-widget', function (hooks) {
+  setupRenderingTest(hooks);
+
+  const ARRAY_OF_MONTHS = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ];
+
+  hooks.beforeEach(function () {
+    this.set('handleClientActivityQuery', sinon.spy());
+    this.set('handleCurrentBillingPeriod', sinon.spy());
+    this.set('arrayOfMonths', ARRAY_OF_MONTHS);
+    this.set('endTimeFromResponse', ['2022', 0]);
+  });
+
+  test('it renders and can open the calendar view', async function (assert) {
+    await render(hbs`
+      <CalendarWidget
+        @arrayOfMonths={{arrayOfMonths}}
+        @endTimeDisplay={{"January 2022"}}
+        @endTimeFromResponse={{endTimeFromResponse}}
+        @handleClientActivityQuery={{handleClientActivityQuery}}
+        @handleCurrentBillingPeriod={{handleCurrentBillingPeriod}}
+        @startTimeDisplay={{"February 2021"}}
+      />
+    `);
+
+    await calendarDropdown.openCalendar();
+    assert.ok(calendarDropdown.showsCalendar, 'renders the calendar component');
+  });
+
+  test('it does not allow a user to click to a future year but does allow a user to click to previous year', async function (assert) {
```

Comment:
It's a little unclear on this one which value passed in it's basing on "previous" vs "future". I think having all the values be different dates might help make that apparent

Reply:
Good idea. I'll amend.